charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml

```
@@ -136,8 +136,9 @@ spec:
 {{- if .Values.prometheus.prometheusSpec.scrapeInterval }}
   scrapeInterval: {{ .Values.prometheus.prometheusSpec.scrapeInterval }}
 {{- end }}
-{{- if .Values.prometheus.prometheusSpec.scrapeProtocols }}
-  scrapeProtocols: {{ .Values.prometheus.prometheusSpec.scrapeProtocols }}
+{{- with .Values.prometheus.prometheusSpec.scrapeProtocols }}
+  scrapeProtocols:
+    {{- tpl (toYaml . | nindent 4) $ }}
```

```suggestion
    {{- toYaml . | nindent 4 }}
```

should suffient as well?

---

correct, local test works the same after the changes